core/src/main/scala/io/iteratee/Enumeratee.scala

```diff
@@ -82,6 +82,34 @@ final object Enumeratee {
       }
     }
 
+  final def tailRecM[F[_], A, B](
+    f: A => Enumerator[F, Either[A, B]]
+  )(implicit F: Monad[F]): Enumeratee[F, Either[A, B], B] =
+    new PureLoop[F, Either[A, B], B] {
+      protected final def loop[C](step: Step[F, B, C]): Step[F, Either[A, B], Step[F, B, C]] =
+        new StepCont[F, Either[A, B], B, C](step) {
+          @annotation.tailrec
+          final def feedEl(e: Either[A, B]): F[Step[F, Either[A, B], Step[F, B, C]]] =
+            e match {
+              case Left(a) =>
+                f(a) match {
+                  case Enumerator.EnumOne(e, _) => feedEl(e)
+                  case other                    => other(this)
```

Comment:
we could also use `F.tailRecM` here if we had `other` that looped into `feedEl` inside `F[_]`:
```scala
def loop(e: Either[A, B]): F[Either[A, B], Step[F, Either[A, B], Step[F, B, C]]]] = 
  e match {
    case Left(a) =>
      f(a) match {
        case Enumerator.EnumOne(e, _) => F.pure(Left(e))
        case ...
     }
   case Right(b) => // same as current
}
```
then `def feedEl(e: Either[A, B]) = F.tailRecM(e)(loop)`

there would be a need to do this if we could remove all or most of the `other(this)` branches by perhap matching on `Enumerator.Map`, `Enumerator.FlatMap`, `Enumerator.FlatMapM` etc... case classes which are likely common transformations of `Enumerator`.